serial.go

```diff
@@ -336,6 +345,149 @@ var defaultMode = &serial.Mode{
 	StopBits: serial.OneStopBit,
 }
 
+const serialSettingsPath = "/userdata/serialSettings.json"
+
+type Terminator struct {
+	Label string `json:"label"` // Terminator label
+	Value string `json:"value"` // Terminator value
+}
+
+type QuickButton struct {
+	Id         string     `json:"id"`         // Unique identifier
+	Label      string     `json:"label"`      // Button label
+	Command    string     `json:"command"`    // Command to send, raw command to send (without auto-terminator)
+	Terminator Terminator `json:"terminator"` // Terminator to use: None/CR/LF/CRLF/LFCR
+	Sort       int        `json:"sort"`       // Sort order
+}
+
+// Mode describes a serial port configuration.
+type CustomButtonSettings struct {
+	BaudRate           string        `json:"baudRate"`           // The serial port bitrate (aka Baudrate)
+	DataBits           string        `json:"dataBits"`           // Size of the character (must be 5, 6, 7 or 8)
+	Parity             string        `json:"parity"`             // Parity (see Parity type for more info)
+	StopBits           string        `json:"stopBits"`           // Stop bits (see StopBits type for more info)
+	Terminator         Terminator    `json:"terminator"`         // Terminator to send after each command
+	LineMode           bool          `json:"lineMode"`           // Whether to send each line when Enter is pressed, or each character immediately
+	HideSerialSettings bool          `json:"hideSerialSettings"` // Whether to hide the serial settings in the UI
+	EnableEcho         bool          `json:"enableEcho"`         // Whether to echo received characters back to the sender
+	Buttons            []QuickButton `json:"buttons"`            // Custom quick buttons
+}
+
+func getSerialSettings() (CustomButtonSettings, error) {
+	config := CustomButtonSettings{
+		BaudRate:           strconv.Itoa(defaultMode.BaudRate),
```

Comment:
Wouldn't it be simpler to store/use the numeric value as integers (baud,data-bits) floats (stop bits)?